plugins/terminal.koplugin/main.lua

```diff
@@ -120,6 +169,63 @@ function Terminal:updateItemTable()
     UIManager:show(self.shortcuts_dialog)
 end
 
+-- other place where placeholders are used: Terminal:showHelp():
+function Terminal:substitutions()
+    for _, v in pairs(self.placeholders) do
+        local placeholder = getPlaceholder(v[1])
+        if self.command:match("%%" .. placeholder) then
+            local substitution = v[2]
+            self.command = substitution(self.command)
+        end
+    end
+end
+
+function Terminal:commandHandler(commands)
+    self.command = self:ensureWhitelineAfterCommands(commands)
+    self:substitutions()
```

Comment:
Please use a verb :)  like `self:substitutePlaceHolders()`
And better passe it the string, which makes it clearer:
`self.command = self:substitutePlaceHolders(self.command)`
and better to make it a local function then
`self.command = substitutePlaceHolders(self.command)`